src/App.tsx

```diff
@@ -1,8 +1,87 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.css';
+import { getData } from './api/getData';
+import { Post, User, Comment } from './components/Interfaces';
+import { PostsList } from './components/PostsList';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+const API = {
+  POSTS: 'https://mate-academy.github.io/react_dynamic-list-of-posts/api/posts.json',
+  USERS: 'https://mate-academy.github.io/react_dynamic-list-of-posts/api/users.json',
+  COMMENTS: 'https://mate-academy.github.io/react_dynamic-list-of-posts/api/comments.json',
+};
 
-export default App;
+export const App = () => {
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const [query, setQuery] = useState<string>('');
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [comments, setComments] = useState<Comment[]>([]);
```

Comment:
если тебе не нужны значения по умолчанию, то не нужно их передавать
```suggestion
  const [query, setQuery] = useState<string>();
  const [posts, setPosts] = useState<Post[]>();
  const [users, setUsers] = useState<User[]>();
  const [comments, setComments] = useState<Comment[]>();
```

Reply:
Якщо не передавати, то typescript свариться